src/brod.erl

```diff
@@ -174,7 +174,19 @@ get_consumer(Client, Topic, Partition) ->
                        | {producer_not_found, topic()}
                        | {producer_not_found, topic(), partition()}.
 get_producer(Client, Topic, Partition) ->
-  brod_client:get_producer(Client, Topic, Partition).
+  case brod_client:get_producer(Client, Topic, Partition) of
+    {ok, Producer} ->
+      {ok, Producer};
+    {error, Reason} = Error
+      when Reason =:= {producer_not_found, Topic};
+           Reason =:= {producer_not_found, Topic, Partition} ->
```

Comment:
this error means topic is found, but there is no such partition, should not  try to start a producer.